src/conduits/RwaOutputConduit3.sol

```diff
@@ -0,0 +1,385 @@
+ *  - The `quit` method allows moving outstanding DAI balance to `quitTo`. It can be called only by `mate`d addresses.
+ *  - The `quit` method with `wad` argument allows moving specified amount of DAI balance to `quitTo`
+ *  - The `file` method allows updating `quitTo` addresses. It can be called only by the admin.
+ */
+contract RwaOutputConduit3 {
+    /// @notice PSM GEM token contract address.
+    GemAbstract public immutable gem;
+    /// @notice PSM contract address.
+    PsmAbstract public immutable psm;
+    /// @dev DAI/GEM resolution difference.
+    uint256 private immutable to18ConvertionFactor;
+
+    /// @notice Addresses with admin access on this contract. `wards[usr]`
+    mapping(address => uint256) public wards;
+    /// @dev Addresses with operator access on this contract. `can[usr]`
+    mapping(address => uint256) public can;
+
+    /// @dev Not used
+    address private __unused_gov;
+    /// @notice Dai token contract address.
+    DaiAbstract public dai;
+    /// @notice Dai Recipient address.
+    address public to;
+
+    /// @notice Whitelist for addresses which can be picked. `bud[who]`
+    mapping(address => uint256) public bud;
+    /// @notice Addresses with push access on this contract. `may[usr]`
+    mapping(address => uint256) public may;
+
+    /// @notice Address to where DAI goes after calling `quit`
+    address public quitTo;
+
+    /**
+     * @notice `usr` was granted admin access.
+     * @param usr The user address.
+     */
+    event Rely(address indexed usr);
+    /**
+     * @notice `usr` admin access was revoked.
+     * @param usr The user address.
+     */
+    event Deny(address indexed usr);
+    /**
+     * @notice `usr` was granted operator access.
+     * @param usr The user address.
+     */
+    event Hope(address indexed usr);
+    /**
+     * @notice `usr` operator access was revoked.
+     * @param usr The user address.
+     */
+    event Nope(address indexed usr);
+    /**
+     * @notice `usr` was granted push access.
+     * @param usr The user address.
+     */
+    event Mate(address indexed usr);
+    /**
+     * @notice `usr` push access was revoked.
+     * @param usr The user address.
+     */
+    event Hate(address indexed usr);
+    /**
+     * @notice `who` address whitelisted for pick.
+     * @param who The user address.
+     */
+    event Kiss(address indexed who);
+    /**
+     * @notice `who` address was removed from whitelist.
+     * @param who The user address.
+     */
+    event Diss(address indexed who);
+    /**
+     * @notice `who` address was picked as the recipient.
+     * @param who The user address.
+     */
+    event Pick(address indexed who);
+    /**
+     * @notice `wad` amount of Dai was pushed to the recipient `to`.
+     * @param to The Dai recipient address
+     * @param wad The amount of Dai
+     */
+    event Push(address indexed to, uint256 wad);
```

Comment:
The actual usage of this event in this contract is for logging the gem amount, not the wad:
`emit Push(recipient, gemAmt)`

Reply:
done